packages/cli/src/execution-lifecycle-hooks/shared/__tests__/shared-hook-functions.test.ts

```diff
@@ -0,0 +1,37 @@
+import { mock } from 'jest-mock-extended';
+import type { IRun } from 'n8n-workflow';
+import { NodeOperationError } from 'n8n-workflow';
+
+import { determineFinalExecutionStatus } from '../shared-hook-functions';
+
+describe('determineFinalExecutionStatus', () => {
```

Comment:
Thanks for adding tests! In future when it's fully covered let's look into simplifying this helper, e.g. I wonder why this old logic is relying on `waitTill` when we have a `waiting` status, etc.